### PR TITLE
Changes for armake

### PR DESCRIPTION
**Pull request by jonpas**
**WIP**
Contains various changes for building on Linux or compatibility with [armake](https://github.com/KoffeinFlummi/armake/).

TODO:
- [x] ~~`diagnostics` component - undefined macros [`INPUT_CTRL_OFFSET` and `INPUT_ALT_OFFSET`](https://github.com/CBATeam/CBA_A3/blob/2bf631362781b7a44198047e309e3c125339ffa8/addons/diagnostic/CfgDisplay3DEN.hpp)~~ `\a3\ui_f\hpp\defineDIKCodes.inc`
- [x] ~~`help` component - undefined macros [`GUI_GRID_W` and `GUI_GRID_H`](https://github.com/CBATeam/CBA_A3/blob/2bf631362781b7a44198047e309e3c125339ffa8/addons/help/script_dialog_defines.hpp)~~ `\a3\ui_f\hpp\defineCommonGrids.inc`
- [x] ~~(maybe intentional?) `​main_a3` and `ui` components - `"true"` instead of `true`/`1` (pboProject automatically changes even quoted variants, armake will not - design decision)~~ intentional
- [x] Add Windows binaries and batch file for creating a full build without `make`
- [x] Git commit hash in bisign name
- [x] Update versions in files (auto-commit, optional push)
- [x] Fix warnings
- [x] Update binaries to armake which includes https://github.com/KoffeinFlummi/armake/pull/65 and use `-e version=<git-hash>`
- [ ] Use `echo -e` for version file updates where newlines are used
- [ ] Use `make.ps1` from @BaerMitUmlaut for auto-downloading armake on Windows
- [ ] Change Makefile as per [here](https://github.com/CBATeam/CBA_A3/pull/730#issuecomment-411018642)

### CODE REVIEW

**Comment by commy2** [comment]
Why "ongoing"?

**Comment by jonpas** [comment]
Unsure, I always tag tools stuff as ongoing, because they mostly don't matter for which release they are in.

Doesn't matter overall, just a habit.

**Comment by commy2** [comment]
So this could be reviewed / merged, right?

**Comment by jonpas** [comment]
Yes, it is finished, CBA fully builds. Check Slack, I wrote about it.

File: VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+3.6.1.180131
```
Comment (Killswitch00) [inline VERSION]: Where is this file used?

**Comment by Killswitch00** [comment]
```
$ make
  PBO  @CBA_A3/addons/cba_hashes.pbo.pbo
error: Failed to copy to temp folder.
make: *** [@CBA_A3/addons/cba_hashes.pbo.pbo] Error 3
```

**Comment by jonpas** [comment]
Back on WIP due to missing header extensions support in armake, when https://github.com/KoffeinFlummi/armake/pull/65 is merged in armake will update binaries here.

File: addons/main/script_version.hpp [hunk 2]
```diff
@@ -1,4 +1 @@
-#define MAJOR 3
-#define MINOR 7
-#define PATCHLVL 0
-#define BUILD 180430
+#define MAJOR 3\n#define MINOR 7\n#define PATCHLVL 0\n#define BUILD 180430
```
Comment (dedmen) [inline addons/main/script_version.hpp]: This looks like a mistake

File: mod.cpp [hunk 3]
```diff
@@ -11,4 +11,4 @@ overview = "What does the name Community Base Addons mean? It is a system that o
 author = "CBA Team";
 overviewPicture = "logo_cba_ca.paa";
 overviewText = "Community Base Addons overviewText";
-overviewFootnote = "<br /><br /><t color='#999999'>This content is under GPLv2 License.<br />Press <t /><t color='#19d3ff'>Left Shift + P<t /><t color='#999999'> to open the store page for more information.<t />";
+overviewFootnote = "<br /><br /><t color='#3.7.0'>This content is under GPLv2 License.<br />Press <t /><t color='#19d3ff'>Left Shift + P<t /><t color='#3.7.0'> to open the store page for more information.<t />";
```
Comment (dedmen) [inline mod.cpp]: That's not a valid color

File: addons/main/script_version.hpp (hunk 2 above)
Comment (SilentSpike) [reply addons/main/script_version.hpp]: Regex? 😆

Comment (dedmen) [reply addons/main/script_version.hpp]: Who is `Roderick Mabry` and how did he get push access?

Comment (ViperMaul) [reply addons/main/script_version.hpp]: This whole thing is quite interesting. 
I was testing out some new merge tools. 
Looks like some fixes are in order.

File: mod.cpp (hunk 3 above)
Comment (ViperMaul) [reply mod.cpp]: Bizarre. Good catch.

File: addons/main/script_version.hpp (hunk 2 above)
Comment (dedmen) [reply addons/main/script_version.hpp]: This is not fixed yet ;) you only fixed the color issue. And now github shows this comment as outdated 🤔

Comment (ViperMaul) [reply addons/main/script_version.hpp]: Micro Commits. Its coming :)

**Comment by dedmen** [comment]
Also integrate the auto-downloading/updating powershell script from ACE armake? instead of including binaries?

**Comment by jonpas** [comment]
Excerpt from Slack, will change workflow as so, to keep it cleaner and less confusing, also more to standard with other big command line publishing tools.
> `release` - clean old files, build all, sign and archive
> `publish` - version update + preparation commit + `release` + push prep commit?

@dedmen yes.

**Comment by dedmen** [comment]
This one:
https://github.com/acemod/ACE3/pull/3747/files/e916f3eb5b9d090133a18bb681430b296a4860db..3ebdb31a821d478b4fbca922eaf44982ed5ad92e#diff-a1349a1002490dc560873829d531b581
And keep in mind
https://github.com/acemod/ACE3/pull/3747#discussion_r192170181

File: .travis.yml [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
 language: python
 python:
-  - '3.5'
+- '3.6'
+before_install:
+- sudo add-apt-repository ppa:koffeinflummi/armake -y
+- sudo apt-get update -qq
+- sudo apt-get install armake -y
+- make --version
+- armake --version
```
Comment (dedmen) [inline .travis.yml]: Shouldn't this be indented one level?

**Comment by jonpas** [comment]
Superseded by HEMTT #1145.

**Pull request closed without merging**